### PR TITLE
Add the DFS and BFS traversal algorithms

### PR DESCRIPTION
**Pull request by jvanecek**
A straightforward implementations of the algorithms to solve the issues #7 and #8

### CODE REVIEW

**Review by fortizpenaloza — CHANGES_REQUESTED** [review]
Improve the docs on https://github.com/ba-st/Mole/blob/master/docs/Mole.md to reflect the changes

File: source/Mole/DirectedGraph.class.st [hunk 1]
```diff
@@ -24,6 +24,13 @@ DirectedGraph class >> withValidVertices: aVertexCollection andEdges: anEdgeColl
 	^ self new initializeWithVertices: aVertexCollection asSet andEdges: anEdgeCollection asSet
 ]
 
+{ #category : #accessing }
+DirectedGraph >> directSuccessorsOf: aSourceVertex [
+
+	^ (self edgesOutgoingFrom: aSourceVertex)
+		collect: [ :directedEdge | directedEdge target ]
```
Comment (fortizpenaloza) [inline source/Mole/DirectedGraph.class.st]:
```smalltalk
^ (self edgesOutgoingFrom: aSourceVertex) collect: #target
```

File: source/Mole/UndirectedGraph.class.st [hunk 2]
```diff
@@ -39,6 +39,15 @@ UndirectedGraph >> degreeOf: aVertex [
...
 ]
 
+{ #category : #accessing }
+UndirectedGraph >> directSuccessorsOf: aVertex [
+
+	^ (self edgesIncidentTo: aVertex) flatCollect: [ :undirectedEdge |
+		undirectedEdge isSelfLoop
+			ifTrue: [ Array with: aVertex ]
+			ifFalse: [ undirectedEdge endpoints copyWithout: aVertex ] ]
+]
```
Comment (fortizpenaloza) [inline source/Mole/UndirectedGraph.class.st]: Can be changed to a simpler collaboration using #reject: ?

**Review by fortizpenaloza — CHANGES_REQUESTED** [review]

File: source/Mole-Tests/GraphTraversalAlgorithmTest.class.st [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+{ #category : #testing }
+GraphTraversalAlgorithmTest class >> isAbstract [
+
+	^ self name = #GraphTraversalAlgorithmTest
+]
+
+{ #category : #private }
+GraphTraversalAlgorithmTest >> directedGraph [
```
[the remark this answers is not in the document]
Comment (jvanecek) [reply source/Mole-Tests/GraphTraversalAlgorithmTest.class.st]: Glad you suggested it as it let me fix a bug in BFS with loops in vertices

**Review by gcotelli — APPROVED** [review]

**Pull request merged**